source/api/commands/route2.md

```diff
@@ -0,0 +1,376 @@
+`path` | `null` | *String, RegExp, Glob* | Request path after the hostname, including query params
+`pathname` | `null` | *String, RegExp, Glob* | Request path after the hostname, without query params
+`port` | `null` | *Number* | Requested port number
+`query` | `null` | *Object* | Query parameters
+`url` | `null` | *String, RegExp, Glob* | Full request URL
+
+**{% fa fa-angle-right %} response** ***(String, Object, Array, Function)***
+
+This allows you to supply a response `body` to {% url 'stub' stubs-spies-and-clocks#Stubs %} in the matching route. This is accomplished by giving you access to the request that you can use in a plain function.
+
+```js
+cy.route2('/users/**', {
+  statusCode: 200,
+  body: {
+    profile: {
+      firstName: 'Tony',
+      lastName: 'Jarvis'
+    }
+  }
+})
+```
+
+In addition, because you have access to the request, this means that you can modify the request as well:
+
+```js
+cy.route2('/users/**', (req) => {
+  req.headers['accept'] = 'application/json'
+  req.body = { ...req.body, note: 'Custom note' }
+})
+```
+
+**{% fa fa-angle-right %} method** ***(String)***
+
+Match the route to a specific HTTP method (`GET`, `POST`, `PUT`, etc).
+
+{% note bolt %}
+By default, Cypress will match `ALL` requests.
+{% endnote %}
+
+## Yields {% helper_icon yields %}
+
+{% yields null_alias cy.route2 %}
+
+# Examples
+
+## Without Stubbing
+
+If you do not pass a `response` to a route, Cypress will pass the request through without stubbing it. We can still wait for the request to resolve later.
+
+### Wait on `GET` request matching `url`
+
+```javascript
+cy.route2('**/users').as('getUsers')
+cy.visit('/users')
+cy.wait('@getUsers')
+```
+
+### Wait on matching `method` and `url`
+
+```javascript
+cy.route2('POST', '**/users').as('postUser')
+cy.visit('/users')
+cy.get('#first-name').type('Julius{enter}')
+cy.wait('@postUser')
+```
+
+### Wait on `url` matching glob
+
+Under the hood Cypress uses {% url 'minimatch' https://github.com/isaacs/minimatch %} to match glob patterns of `url`.
+
+This means you can take advantage of `*` and `**` glob support. This makes it *much* easier to route against dynamic segments without having to build up a complex `RegExp`.
+
+We expose {% url `Cypress.minimatch` minimatch %} as a function that you can use in your console to test routes.
+
+### Match route against any UserId
+
+```javascript
+cy.route2('**/users/*/comments')
+
+// https://localhost:7777/users/123/comments     <-- matches
+// https://localhost:7777/users/123/comments/465 <-- does not match
+```
+
+### Use glob to match all segments
+
+```javascript
+cy.route2('**/posts/**')
+
+// https://localhost:7777/posts/1            <-- matches
+// https://localhost:7777/posts/foo/bar/baz  <-- matches
+// https://localhost:7777/posts/quuz?a=b&1=2 <-- matches
+```
+
+### Modify request
+
+Before a request gets passed through, we can modify the request. The `response` function accepts the request object as a parameter.
```

Comment:
it's confusing to call this the `response` function. that's the parameter's name, but the type is a `RouteHandler` in the code, which narrows to a `HttpRequestInterceptor` if it's a function. so 

```suggestion
Before a request gets sent to the real endpoint, we can modify the request. The supplied `RouteHandler` callback receives the request object as a parameter.
```

another place where it would be good to have links to types